packages/@o3r/core/src/utils/deep-fill/deep-fill.ts

```diff
@@ -66,9 +66,11 @@ export function deepFill<T extends { [x: string]: any }>(base: T, source?: { [x:
   }
   const newObj = {...base};
   for (const key in base) {
-    if (key in source && typeof base[key] === typeof source[key]) {
+    if (key in source && typeof base[key] === typeof source[key] && source[key] !== null) {
```

Comment:
you don't need to change this condition if you move the `else if ` block up
```suggestion
    if (source[key] === null) {
      newObj[key] = immutablePrimitive(null, additionalMappers);
    } else if (key in source && typeof base[key] === typeof source[key]) {
```